### PR TITLE
Add `CARGO_MAKE_CRATE_TARGET_DIR` env variable

### PR DESCRIPTION
**Pull request by ark0f**

### CODE REVIEW

File: src/lib/environment/crateinfo.rs [hunk 1]
```diff
@@ -187,48 +186,62 @@ pub(crate) fn load_from(file_path: PathBuf) -> CrateInfo {
...
+    let path = env::current_dir().ok()?;
 
-        for config_folder in config_folders {
-            let config_file = config_folder.join("config");
+    let config_file = path
+        .ancestors()
+        .map(|ancestor| ancestor.join(".cargo"))
+        .chain(home)
+        .map(|config_file| config_file.join("config"))
+        .filter_map(|config_file| {
```
Comment (sagiegurari) [inline src/lib/environment/crateinfo.rs:209]: i'm guessing the regression might be here, but didn't investigate.

**Review by sagiegurari — CHANGES_REQUESTED** [review]
@ark0f thanks a lot for the PR. this env var is actually super useful.
i see a test is failing so maybe worth to check what is the regression there.

**Comment by sagiegurari** [comment]
@ark0f also i merged your other PR, so you need to resolve merge conflicts

**Comment by sagiegurari** [comment]
@ark0f looks awesome. lets wait for the CI to finish and than i'll merge it.
thanks a lot!

**Review by sagiegurari — APPROVED** [review]

**Comment by ark0f** [comment]
All tests pass after I added `#[ignore]` to `get_crate_target_dir` test (like previous contributor did with `get_crate_target_triple`), but I still don't know why changing CWD can affect tests

**Comment by sagiegurari** [comment]
> I still don't know why changing CWD can affect tests

running cargo test will run tests in parallel so changing cwd will break other tests that look at the file system.
when you add ignore to a test, i postpone it to later and run it in a single thread run (cargo make splits and runs its tests in 2 batches, parallel and single threaded).

**Comment by ark0f** [comment]
Oops, forgot about import

**Comment by sagiegurari** [comment]
looks great. thanks again. merging

**Pull request merged**